jubilant/_juju.py

```diff
@@ -837,6 +857,142 @@ def scp(
+
+    @overload
+    def show_secret(
+        self,
+        name_or_uri: str | SecretURI,
+        /,
+        *,
+        reveal: Literal[True],
+        revision: int | None = None,
+        revisions: Literal[False] = False,
+    ) -> Secret[Revealed]: ...
+
+    @overload
+    def show_secret(
+        self,
+        name_or_uri: str | SecretURI,
+        /,
+        *,
+        reveal: Literal[False] = False,
+        revision: int | None = None,
+        revisions: Literal[False] = False,
+    ) -> Secret[Hidden]: ...
+
+    @overload
+    def show_secret(
+        self,
+        name_or_uri: str | SecretURI,
+        /,
+        *,
+        reveal: Literal[False] = False,
+        revision: None = None,
+        revisions: Literal[True],
+    ) -> Secret[Hidden]: ...
+
+    def show_secret(
+        self,
+        name_or_uri: str | SecretURI,
+        /,
+        *,
+        reveal: bool = False,
+        revision: int | None = None,
+        revisions: bool = False,
+    ) -> Secret[Revealed] | Secret[Hidden]:
+        """Get the content of a secret.
+
+        Args:
+            name_or_uri: Name or URI of the secret to reveal.
+            reveal: Whether to reveal the secret content.
+            revisions: Whether to include all revisions of the secret. Mutually
+                exclusive with `reveal` and `revision`.
+            revision: Optional revision number of the secret to reveal. If not specified,
+                the latest revision is revealed.
+        """
+        args = ['show-secret', name_or_uri, '--format', 'json']
+        if reveal:
+            args.append('--reveal')
+        if revisions:
+            args.append('--revisions')
+        if revision is not None:
+            args.extend(['--revision', str(revision)])
+        stdout = self.cli(*args)
+        output: dict[str, _SecretResponse] = json.loads(stdout)
+        uri_from_juju, obj = next(iter(output.items()))
+        return Secret[Any](
+            uri=SecretURI('secret:' + uri_from_juju),
+            name=obj.get('name'),
+            label=obj.get('label'),
+            owner=obj.get('owner'),
+            rotates=datetime.datetime.fromisoformat(obj['rotates'].replace('Z', '+00:00'))
+            if 'rotates' in obj
+            else None,
+            rotation=obj.get('rotation'),
+            expires=obj.get('expires'),
+            revision=obj.get('revision'),
+            description=obj.get('description', ''),
+            created=datetime.datetime.fromisoformat(obj['created'].replace('Z', '+00:00')),
+            updated=datetime.datetime.fromisoformat(obj['updated'].replace('Z', '+00:00')),
+            content=obj.get('content', {}).get('Data', None),
+            checksum=obj.get('checksum', ''),
+            access=[
+                SecretAccess(
+                    role=SecretAccessRole(access.get('role')),
+                    scope=SecretAccessScope(access.get('scope')),
+                    target=access.get('target'),
```

Comment:
Let's move this to a `SecretAccess._from_dict` helper (same for `SecretRevision._from_dict`), similar to how we parse the big `Status` object.